src/addrman.h

```diff
@@ -295,8 +303,17 @@ friend class CAddrManTest;
 
     /**
      * Serialized format.
-     * * version byte (@see `Format`)
-     * * 0x20 + nKey (serialized as if it were a vector, for backward compatibility)
+     * * format version byte (@see `Format`)
+     * * lowest compatible format version byte. This is used to help old software decide
+     *   whether to parse the file. For example:
+     *   * Bitcoin Core 0.21 knows how to parse up to format=3. If a new format=4 is introduced
+     *     in Bitcoin Core 0.23 which is compatible with format=3 and it is known that 0.21 will
+     *     be able to parse it, then 0.23 will write (format=4, lowest_compatible=3) in the
+     *     first two bytes of the file and so 0.21 will still try to parse it.
+     *   * Bitcoin Core 0.25 introduces a new incompatible format=5. It will write
+     *     (format=5, lowest_compatible=5) and so any versions that do not know how to parse
+     *     format=5 will not try to read the file.
```

Comment:
I'd suggest not using actual version numbers in this comment. At some point we'll have versions 0.23 and 0.25, and they might not use file versions 4 and 5, and this comment will be confusing.

Reply:
Changed to `V`, `V+1` and `V+2`.